src/state.rs

```diff
@@ -55,40 +59,118 @@ impl Persistent {
     pub fn password_set(&self) -> bool {
         self.authorization_key.is_some()
     }
+    pub fn get_kek<T>(&mut self, trussed: &mut T) -> trussed::error::Result<KeyId>
+    where
+        T: trussed::Client + trussed::client::Chacha8Poly1305,
+    {
+        Ok(match self.kek {
+            None => {
+                let r = try_syscall!(trussed.generate_chacha8poly1305_key(Location::Internal))?.key;
+                self.kek = Some(r);
```

Comment:
Currently there is no other use of the Persistent type other, then through the State, which loads and saves it immediately on each use. See `State::get_kek()`:

https://github.com/Nitrokey/oath-authenticator/pull/16/files/1b31c9a87236fcf27574f918bdfbf3c5aabbaecd#diff-87619baf236177a7a9fc38150961151b7583b6f2f082ade75915f3ac4ca49c41R104-R110

While I believe this is safe now, it generates a lot of writes unnecessarily (I doubt these are optimized away on any level), and from this perspective surely the refactor would be needed.

Reply:
I don't believe it is safe. For example:
- a new credential is generated
- `get_kek` generates the key (kek1)
- the credential is stored encrypted with kek1
- the app crashes (our the key is removed)
- ~~the state (including the kek1 id) is stored~~ This does not happen because of the crash

Now the state and the encrypted credentials are out of sync.

Reply:
But in the current implementation the KeyID is saved just after step 2 in the persistent storage, before the step 3 happens.

Reply:
My bad. There are two `get_kek`, one on the state and one on the persistent state. I missed the first.

Reply:
I thought so. Let me rename these two appropriately to remove this confusion.